File: backend/bar/event.py
from xml.dom.minidom import parseString
import pandas as panda
import backend.general_functions as general_functions

panda.set_option('display.max_rows', None)
panda.set_option('display.max_columns', None)
panda.set_option('display.width', None)
panda.set_option('display.max_colwidth', None)



def sort_new_DataFrame(dictionary, DataFrame, column_dictionary):

    #Note ((dictionary["ID of Dropdown"] == "Address") and not dictionary["Chosen Addresses"]) means that they manually deselected the address's themselves
    #Note Meaning the "Address" dropdown is empty, so we just return the default DataFrame
    if (dictionary["ID of Dropdown"] == "Type") or (dictionary["ID of Dropdown"] == "Reset-Icon") or ((dictionary["ID of Dropdown"] == "Address") and not dictionary["Chosen Addresses"]):
        return DataFrame

    elif (dictionary["ID of Dropdown"] == "Address") and dictionary["Chosen Addresses"]:
            return general_functions.sort_DataFrame_for_Addresses(dictionary["Chosen Addresses"], DataFrame, column_dictionary["Address Column"])

    elif (dictionary["ID of Dropdown"] == "Less-Than") or (dictionary["ID of Dropdown"] == "Greater-Than"):
        return general_functions.sort_inequality_DataFrame(dictionary, DataFrame, column_dictionary["Inequality Column"])

               

File: frontend/static/js/bar/event.js
import {repopulate_address_dropdown, repopulate_inequality_dropdowns, reset_selected_inequality} from "../dropdown_functions.js"

function event_checker(event_id, result)
{
    if (event_id == "Type")
        {type_event(result)}

    else if (event_id == "Address")
        {address_event(result)}

    //Note Reset the selected values for inequality dropdown
    else if (event_id == "Reset-Icon")
        {reset_selected_inequality()}
}

function type_event(result)
{
    $("#Less-Than").empty();
    $("#Greater-Than").empty();
    repopulate_inequality_dropdowns("#Less-Than", result["Descending List"])
    repopulate_inequality_dropdowns("#Greater-Than", result["Ascending List"])
}

function address_event(result)
{
    //Note Check if the key "Chosen Addresses" is in dropdown_dictionary
    if (("Chosen Addresses" in dropdown_dictionary))
    {
        dropdown_dictionary["Chosen Addresses"].split(',');
        repopulate_address_dropdown(result["Address List"], dropdown_dictionary["Chosen Addresses"])
        
    }
    else
    {
        for(let i = 0; i < result["Address List"].length; i++) 
        {
            $("#Address").append($("<option>").val(result["Address List"][i]).text(result["Address List"][i]));        
        }
    }
}

export {event_checker}